py/digamma.py

```diff
@@ -0,0 +1,82 @@
+#! /usr/bin/env python
+import numpy as np
+from numba import vectorize
+
+@vectorize(nopython=True)
+def digamma(x):
+
+#*****************************************************************************80
+#
+## DIGAMMA calculates DIGAMMA ( X ) = d ( LOG ( GAMMA ( X ) ) ) / dX
+#
+#  Licensing:
+#
+#    This code is distributed under the GNU LGPL license.
```

Comment:
I think we might be stuck using the GPL for this library anyway because Rmath is GPL.

However, if there is a way around being forced into the GPL because of Rmath, it would be great for us to write our own version of this routine that isn't GPL.

@jstac what do you think?